pims/base_frames.py

```diff
@@ -8,6 +8,9 @@
 import numpy as np
 import collections
 import itertools
+import functools
+import contextlib
+from threading import Lock
```

Comment:
I saw that the Cine code was meant to be thread safe, so I just followed that example. I take it that this was never a project-wide goal? If not, perhaps we should replace all the locking stuff with `# NOTE: Not thread-safe`.